src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -164,6 +164,14 @@ protected void createAndRouteAccountingDocumentFromXml(AccountingXmlDocumentEntr
             reportDetail.setSuccessfullyRouted(true);
             reportDetail.setDocumentNumber(document.getDocumentNumber());
             reportItem.getDocumentsSuccessfullyRouted().add(reportDetail);
+            if (!GlobalVariables.getMessageMap().getWarningMessages().isEmpty()
+                    && GlobalVariables.getMessageMap().getWarningMessages()
+                            .containsKey(CuFPConstants.CreateAccountingDocumentValidatedDataElements.PAYEE_NAME)) {

```

Comment:
Is there a reason why the warning messages need to be filtered based on a static field name? If so, could this be configured to allow for deriving the appropriate filter based on the document generator, as opposed to hard-coding the field name at this level?

Reply:
Thanks for the idea, I have moved this logic to CuDisbursementVoucherDocumentGenerator.  Currently, we only care about the warning message generated with this code change, so I only want to write out that warning message.  I added a log entry if there are other warning messages, and we can review those results and make adjustments as needed.